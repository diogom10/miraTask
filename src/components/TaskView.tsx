import React from 'react';
import {Task} from '../models/Task';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {MenuIcon} from '../assets/icons/menu';
import {CheckIcon} from '../assets/icons/check';

interface ITaskViewProps {
  task: Task;
  onPressCard?: () => void;
  onPressOptions?: () => void;
}

export const TaskView = ({
  task,
  onPressCard,
  onPressOptions,
}: ITaskViewProps) => (
  <TouchableOpacity
    activeOpacity={1}
    style={styles.container}
    onPress={onPressCard}>
    <View style={styles.mainSection}>
      <Text style={styles.title} ellipsizeMode="tail" numberOfLines={2}>
        {task?.title}
      </Text>
      <>
        {!task?.completed && (
          <TouchableOpacity style={styles.ctnButton} onPress={onPressOptions}>
            <MenuIcon width={20} height={20} fill={'gray'} />
          </TouchableOpacity>
        )}
      </>
      <>
        {task?.completed && <CheckIcon width={20} height={20} fill={'green'} />}
      </>
      {/*<Text style={styles.check}>{task?.completed ? '✅' : '▢'}</Text>*/}
    </View>
    <Text>{task?.description}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    borderColor: '#dedede',
    borderRadius: 20,
    borderWidth: 1,
    padding: 4,
    margin: 2,
    paddingHorizontal: 10,
    paddingVertical: 8,
    backgroundColor: '#fbfbfb',
    marginBottom: 10,
  },
  mainSection: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    minHeight: 20,
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    width: '92%',
  },
  check: {
    fontSize: 24,
    minHeight: 20,
    width: '8%',
    textAlign: 'center',
  },
  ctnButton: {
    width: 30,
    height: 20,
  },
});
