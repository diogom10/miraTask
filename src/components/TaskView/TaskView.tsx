import React from 'react';
import {Task} from '../../models/Task';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {MenuIcon} from '../../assets/icons/menu';
import {CheckIcon} from '../../assets/icons/check';

export interface ITaskViewProps {
  task: Task;
  onPressCard?: () => void;
  onPressOptions?: () => void;
  testID?: string;
}

export const TaskView = ({
  task,
  onPressCard,
  onPressOptions,
  testID = 'TaskView',
}: ITaskViewProps) => (
  <TouchableOpacity
    activeOpacity={1}
    testID={`${testID}_btn_card`}
    style={styles.container}
    onPress={onPressCard}>
    <View style={styles.mainSection}>
      <Text
        testID={`${testID}_title`}
        style={styles.title}
        ellipsizeMode="tail"
        numberOfLines={2}>
        {task?.title}
      </Text>
      <>
        {!task?.completed && (
          <TouchableOpacity
            testID={`${testID}_btn_options`}
            style={styles.ctnButton}
            onPress={onPressOptions}>
            <MenuIcon width={20} height={20} fill={'gray'} />
          </TouchableOpacity>
        )}
      </>
      <>
        {task?.completed && (
          <CheckIcon
            testID={`${testID}_check_icon`}
            width={20}
            height={20}
            fill={'green'}
          />
        )}
      </>
    </View>
    <Text testID={`${testID}_description`}>{task?.description}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    borderColor: '#dedede',
    borderRadius: 20,
    borderWidth: 1,
    padding: 4,
    margin: 2,
    paddingHorizontal: 10,
    paddingVertical: 8,
    backgroundColor: '#fbfbfb',
    marginBottom: 10,
  },
  mainSection: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    minHeight: 20,
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    width: '92%',
  },
  check: {
    fontSize: 24,
    minHeight: 20,
    width: '8%',
    textAlign: 'center',
  },
  ctnButton: {
    width: 30,
    height: 20,
  },
});
