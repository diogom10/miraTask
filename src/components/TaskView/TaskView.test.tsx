import React from 'react';
import {ITaskViewProps, TaskView} from './TaskView';
import {create, act} from 'react-test-renderer';

const props: ITaskViewProps = {
  task: {
    id: '34c38a41-dd29-468d-bf58-f9ee16214b58',
    title: 'pede justo eu massa donec dapibus duis',
    description:
      'vel enim sit amet nunc viverra dapibus nulla suscipit ligula in lacus curabitur at ipsum ac tellus semper interdum mauris ullamcorper purus sit amet nulla quisque arcu libero rutrum',
    completed: false,
  },
  onPressCard: jest.fn(),
  onPressOptions: jest.fn(),
  testID: 'TaskView',
};

const component = create(<TaskView {...props} />);
const el_title = `${props.testID}_title`;
const el_description = `${props.testID}_description`;
const el_btn_card = `${props.testID}_btn_card`;
const el_btn_options = `${props.testID}_btn_options`;
const el_check_icon = `${props.testID}_check_icon`;

describe('TaskView Component', () => {
  test(`should render btn #${el_btn_options}`, async () => {
    const button = component.root.findByProps({testID: el_btn_options}).props;
    act(() => button.onPress());
  });
  test(`should render label #${el_title}`, async () => {
    const label = component.root.findByProps({testID: el_title}).props;
    expect(label).toBeTruthy();
  });
  test(`should render label #${el_description}`, async () => {
    const label = component.root.findByProps({testID: el_description}).props;
    expect(label).toBeTruthy();
  });
  test(`should render btn #${el_btn_card}`, async () => {
    const button = component.root.findByProps({testID: el_btn_card}).props;
    act(() => button.onPress());
  });
  test(`should render icon #${el_check_icon} when completed is true`, async () => {
    props.task.completed = true;
    const updateComponent = create(<TaskView {...props} />);
    const icon = updateComponent.root.findByProps({
      testID: el_check_icon,
    }).props;
    expect(icon).toBeTruthy();
  });
  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
