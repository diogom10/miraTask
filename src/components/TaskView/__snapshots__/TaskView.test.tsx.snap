// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskView Component snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#fbfbfb",
      "borderColor": "#dedede",
      "borderRadius": 20,
      "borderWidth": 1,
      "margin": 2,
      "marginBottom": 10,
      "opacity": 1,
      "padding": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
  testID="TaskView_btn_card"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 20,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
          "width": "92%",
        }
      }
      testID="TaskView_title"
    >
      pede justo eu massa donec dapibus duis
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 20,
          "opacity": 1,
          "width": 30,
        }
      }
      testID="TaskView_btn_options"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="20"
        bbWidth="20"
        fill="gray"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={512}
        vbWidth={512}
        width={20}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4286611584,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="256"
            cy="256"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            r="48"
          />
          <RNSVGCircle
            cx="256"
            cy="416"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            r="48"
          />
          <RNSVGCircle
            cx="256"
            cy="96"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            r="48"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <Text
    testID="TaskView_description"
  >
    vel enim sit amet nunc viverra dapibus nulla suscipit ligula in lacus curabitur at ipsum ac tellus semper interdum mauris ullamcorper purus sit amet nulla quisque arcu libero rutrum
  </Text>
</View>
`;
